src/Shared/NativeMethodsShared.cs

```diff
@@ -546,14 +546,21 @@ internal static bool IsWindows
             get { return RuntimeInformation.IsOSPlatform(OSPlatform.Windows); }
 #endif
         }
-
+        
+#if MONO
+        private static bool? _isOSX;
```

Comment:
Wouldn't it be better to make a `Lazy<bool>` which calls [this](https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Platform.cs#L67-L84)?

Reply:
I suppose a pinvoke is faster than IO, and lazy will also prevent multiple evaluation.

Reply:
I just wanted a minimal patch which can be easily/safely backported to release branches. If we want a more invasive patch it might be frowned on when we go to backport it

Reply:
(also, multiple execution is not something that we need to be particularly worried about in this case, so i left it as a simple nullable bool)